packages/suspense/src/cache/createCache.ts

```diff
@@ -148,6 +148,16 @@ export function createCache<Params extends Array<any>, Value>(
     return new Map();
```

Comment:
@bvaughn One thing I forgot on final pass is this function. Should this function also use the CacheMap? I don't really understand how `getCacheForType` works, so I'm a little hesitant to change anything.

Reply:
Yes. If this type returned a normal `Map`, it would prevent GC. The `getCache` configuration option should be the thing passed to `getCacheForType`.

Reply:
hey @bvaughn, i've got a couple questions related to this:

1) Should these be kept in sync? If `recordMap` invalidates, should this be propagated to `backupRecordMap` and vice versa?
2) What is the relationship between these two caches? I've read the comments but still don't understand 😅

Reply:
@cevr

I explain the way the two caches work in this Loom:
https://www.loom.com/share/55b645c9730a4958b8a2f91dfc19b72f

As for keeping the two "in sync" – the logic that was in `createCache` and `useCacheMutation` already should be preserved. It was intentional.

Reply:
Ahh I see! 

It seems like the map would only have records that are being mutated no? Is it fair to say it will only ever have a subset of what is in the `backupRecordMap`?

Reply:
In almost every case, this would be a subset, yes.

Keep in mind there may be multiple copies of these (each a non-overlapping subset).

Reply:
It seems to me that the `backupRecord` map is less of a backup and more like the primary source of truth. 

If it uses a subset, then it might make sense to have it bypass the user defined `CacheMap`, since the eviction policies  would most likely be useless for the subsets. Otherwise, we risk conflicting evictions and maybe double notifications? Though it might be a non-issue since react would bail.

Tangentially I'm interested in the use case. Are these cache subsets primarily for notifying of pending mutations to components who did not initiate it? I've never personally have felt the need to know of pending mutations outside of the component, since if it fails it usually has no consequences in regards to the currently cached data. The flow I'm used to is something like mutate -> onSuccess -> (invalidate or update in-place) -> notify. The component would have easy access to tracking the status of the mutation since its in scope.

Reply:
Fair enough!

If the `backupMap` is the source of truth, should we just notify whenever that evicts, or notify for both?

Reply:
That's a good question.

Do we care about evictions in the temporary subset caches? I'm not sure. We don't want to double-notify.

It probably won't matter much in practice. The only case where it would matter (I think?) would be a mutation that starts, but never finishes (or fails)? Then the record would only be in the "smaller" map?

Reply:
Yeah, agreed - it's probably unlikely to happen.

What do you think about renaming `backupRecordMap` to `recordMap` and the current `createRecordMap` to something like `createPendingMutationRecordMap`?

Reply:
This actually does cause an issue with the WeakRefMap. It seems like it creates a closure that react controls that prevents it from ever being fully collected 😩

Reply:
> What do you think about renaming backupRecordMap to recordMap and the current createRecordMap to something like createPendingMutationRecordMap?

I think that sounds fine.

> This actually does cause an issue with the WeakRefMap. It seems like it creates a closure that react controls that prevents it from ever being fully collected 😩

Really 🤔  That's unexepcted.